### PR TITLE
Update version for spring boot starter parent

### PR DESCRIPTION
**Pull request by chenrujun**
Update spring-boot-starter-parent's version to 2.3.5 by command 'python eng/versioning/update_versions.py --ut external_dependency --sr'.

### CODE REVIEW

File: sdk/spring/scripts/replace_util.py [hunk 1]
```diff
@@ -21,56 +21,56 @@
     'appconfiguration': {
         'sdk/appconfiguration/azure-spring-cloud-test-appconfiguration-config/pom.xml': {
             VERSION_UPDATE_ITEMS: (
-                VersionUpdateItem('org.springframework.boot:spring-boot-starter-parent', '2.2.9.RELEASE'),
+                VersionUpdateItem('org.springframework.boot:spring-boot-starter-parent', '2.2.11.RELEASE'),
```
Comment (chenrujun) [inline sdk/spring/scripts/replace_util.py:24]: @jialindai , @saragluna , @yiliuTo , @ZhijunZhao , FYI:

Even though we claim that our starter can support `2.2.x` and `2.3.x` at the same time.
But in this case, `2.2.11` work well, but `2.2.9` will have error.

It's caused by netty/reactor-netty does not backward compatible even for it's minor version up.

**Review by yiliuTo — APPROVED** [review]
LGTM

**Pull request merged**